src/netmap.js

```diff
@@ -1,10 +1,69 @@
 /* TODO: Update what this scripts intention is
  */
 
+const OUT_FILE = 'netmap-data.json';
+const FLAGS_DEF = [['help', false, 'Displays this help message']];
+
+/**
+ * @param {import(".").NS} ns Use just "@param {NS} ns" if editing in game
+ */
+function displayHelp(ns) {
+  // Find the max lengths to help format to the screen
+  const maxLengths = { key: -1, value: -1, description: -1 };
+  const getGreaterLength = (currentMax, itemToCheck) => {
+    if (itemToCheck === null || itemToCheck === undefined) return currentMax;
+    switch (typeof itemToCheck) {
+      case 'boolean':
+        if (itemToCheck) {
+          return currentMax > 4 ? currentMax : 4; // "true"
+        } else {
+          return 5; // "false"
+        }
+      case 'number':
+        const asString = itemToCheck.toString();
+        return currentMax > asString.length ? currentMax : asString.length;
+      case 'string':
+        return currentMax > itemToCheck.length
+          ? currentMax
+          : itemToCheck.length;
+      default:
+        throw new Error(`Missing mapping for type: ${typeof itemToCheck}`);
+    }
+  };
+
+  for (let i = 0; i < FLAGS_DEF.length; i++) {
+    const item = FLAGS_DEF[i];
+    maxLengths.key = getGreaterLength(maxLengths.key, item[0]);
+    maxLengths.value = getGreaterLength(maxLengths.key, item[1]);
+    maxLengths.description = getGreaterLength(maxLengths.key, item[2]);
+  }
+
+  // Print the help
+  ns.tprint(
+    'This script collects information about the currently viewable network.',
+  );
+  ns.tprint('Flags:');
+  for (let i = 0; i < FLAGS_DEF.length; i++) {
+    const item = FLAGS_DEF[i];
+    const [padKey, padVal, padDes] = [
+      item[0].toString().padEnd(maxLengths.key, '.'),
+      item[1].toString().padStart(maxLengths.value, '.'),
+      item[2] || '',
+    ];
+
+    ns.tprint(`  ${padKey} -- default ${padVal} -- ${padDes}`);
+  }
+  ns.tprint(`Usage: run ${ns.getScriptName()}`);
+  ns.tprint('Example:');
+  ns.tprint(`> run ${ns.getScriptName()}`);
+}
```

Comment:
Doh. Probably should of moved this to the "common" implementation too. Didn't notice that until I created the PR and began adding comments for the reviewer.